common/web/keyboard-processor/src/text/kbdInterface.ts

```diff
@@ -882,7 +882,8 @@ namespace com.keyman.text {
      */
     setStore(systemId: number, strValue: string, outputTarget: OutputTarget): boolean {
       this.resetContextCache();
-      if(systemId == KeyboardInterface.TSS_LAYER) {
+      // Unique case:  we only allow set-store ops from keyboard rules triggered by touch OSKs.
+      if(systemId == KeyboardInterface.TSS_LAYER && this.activeDevice.touchable) {
```

Comment:
I take it `this.activeDevice` is correct even when emulating e.g. in the Keyman Developer Server emulated touch keyboard mode?

Reply:
Yes.  Your generalized `process` method for semi-sandboxing the keyboard calls works pretty well for that, so long as it's given the right parameters.  Its setup seems pretty sound.